generate_workspace/src/main/java/com/google/devtools/build/workspace/GenerateWorkspace.java

```diff
@@ -69,13 +79,22 @@ public static void main(String[] args) throws InterruptedException {
     }
   }
 
-  private GenerateWorkspace(String[] args, String outputDirStr, boolean directToWorkspace, List<Rule> aliases)
-      throws IOException {
-    this.resolver = new Resolver(new DefaultModelResolver(), aliases);
+  private GenerateWorkspace(
+    String[] args,
+    String outputDirStr,
+    boolean directToWorkspace,
+    List<String> repositories,
+    List<Rule> aliases
+  ) throws IOException {
+    DefaultModelResolver defaultModelResolver = new DefaultModelResolver();
+    for (String repo : repositories) {
+      defaultModelResolver.addUserRepository(repo);
```

Comment:
Perhaps, overloading the constructor? Admittedly, the base constructor has four arguments as is, so I can see why you would explicitly add the repositories like this.